core/src/main/java/io/ap4k/SessionWriter.java

```diff
@@ -23,32 +24,44 @@
   /**
    * Writes all {@link Session} resources.
    * @param session The target session.
+   * @return Map containing the file system paths of the output files as keys and their actual content as the values
    */
-  default void write(Session session) {
+  default Map<String, String> write(Session session) {
+    final Map<String, String> result = new HashMap<>();
     session.close();
     Map<String, KubernetesList> resources = session.getGeneratedResources();
     Set<? extends Configuration> configurations = session.configurators().toSet();
-    resources.forEach((g, l) -> write(g, l));
-    configurations.forEach(c -> write(c));
-    write(getProject());
+    resources.forEach((g, l) -> {
+      result.putAll(write(g, l));
+    });
+    configurations.forEach(c -> {
+      final Map.Entry<String, String> entry = write(c);
+      result.put(entry.getKey(), entry.getValue());
+    });
+    final Map.Entry<String, String> projectEntry = write(getProject());
+    result.put(projectEntry.getKey(), projectEntry.getValue());
+    return result;
   }
 
   /**
    * Writes a {@link Configuration}.
    * @param config  The target session configurations.
+   * @return Map Entry containing the file system path of the written configuration and the actual content as the value
    */
-   void write(Configuration config);
+  Map.Entry<String, String> write(Configuration config);
```

Comment:
Given that the session itself holds such a map, I don't see much point in also returning it here.

The only differennce I see is that one is a Map<String, KubernetesList> and the other is Map<String ,String>. I'd rather have the Session use a Map<String, String> instead.